src/decorators/BIPComponentTypeDecorator/DiagramDesigner/BIPComponentTypeDecorator.DiagramDesignerWidget.js

```diff
@@ -138,36 +226,43 @@ define([
     };
 
     BIPComponentTypeDecorator.prototype._renderOwnProperties = function () {
-        var client = this._control._client,
-            nodeObj = client.getNode(this._metaInfo[CONSTANTS.GME_ID]),
-            membersInfo = [];
+        var self = this,
+            client = self._control._client,
+            gmeID = self._metaInfo[CONSTANTS.GME_ID],
+            nodeObj = client.getNode(gmeID),
+            validSetNames,
+            patterns = {};
 
         //render GME-ID in the DOM, for debugging
-        this.$el.attr({'data-id': this._metaInfo[CONSTANTS.GME_ID]});
+        this.$el.attr({'data-id': gmeID});
 
         if (nodeObj) {
+            validSetNames = nodeObj.getValidSetNames();
             this.name = nodeObj.getAttribute(nodePropertyNames.Attributes.name) || '';
             this.cardinality = nodeObj.getAttribute(nodePropertyNames.Attributes.cardinality) || '';
             //this.position = nodeObj.getEditableRegistry('position');
             // Use hostDesignerItem's position instead.
             this.position.x = this.hostDesignerItem.positionX;
             this.position.y = this.hostDesignerItem.positionY;
             this.isCompound = this._isOfMetaTypeName(nodeObj.getMetaTypeId(), 'CompoundType');
-            if (this.setIsDefined) {
-                nodeObj.getMemberIds(nodePropertyNames.Sets.styleBases)
+            if (validSetNames.indexOf(nodePropertyNames.Sets.associatedWith) > -1 || validSetNames.indexOf(nodePropertyNames.Sets.implementedBy) > -1) {
+                if (validSetNames.indexOf(nodePropertyNames.Sets.associatedWith) > -1) {
+                    nodeObj.getMemberIds(nodePropertyNames.Sets.associatedWith)
+                      .forEach(function (id) {
+                            patterns[id] = {children: 0};
+                        });
+                } else {
```

Comment:
Collect all member nodes and add them to the `patterns` - otherwise they might never be available in the `eventHandler`.